### ISSUE TITLE
shallowMount fails with TypeError when component uses a data of type `Symbol` in it's template

### ISSUE THREAD

**Issue by penguintree** [vuejs/vue-test-utils#1833]
### Version
1.1.4

### Reproduction link
[https://github.com/penguintree/vue-test-util-symbol-issue](https://github.com/penguintree/vue-test-util-symbol-issue)

### Steps to reproduce
`npm run test:unit`

### What is expected?
Unit tests should pass

### What is actually happening?
One of the unit test fails with error :
`TypeError: Failed to execute 'setAttribute' on 'Element': parameter 2 is a symbol, which cannot be converted to a string.`

-----
Possibly a regression in regards of [this closed issue](https://github.com/vuejs/vue-test-utils/issues/833#issue-342509711)

Basically, it is not possible to shallowMount a component wich passes a Symbol to a child component.

**Comment by phloose** [comment]
Can confirm, It's still an issue and only happening when running tests. In the browser console no errors are shown regarding the usage of a Symbol

**Comment by boboldehampsink** [comment]
Any news on this? Still happening

**Comment by jrummell** [comment]
I can confirm this issue. If I replace `shallowMount` with `mount`, it works.

**Comment by liamdebeasi** [comment]
Some of our users are also running into this issue with Ionic Vue (see: https://github.com/ionic-team/ionic-framework/issues/24181). Vue 3.1 sets all property keys in the properties object, even if certain properties are not actually being used by the end developer (see https://github.com/vuejs/vue-next/issues/3889). We use `Symbol` to differentiate between properties that are used and properties that are not used.

Using `shallowMount` on Ionic Vue components causes this bug to appear. Using `mount` instead of `shallowMount` works for now, but it would be great if this could get fixed.

**Comment by lmiller1990** [comment]
I looked into this, I posted my findings and some ideas to proceed here: https://github.com/ionic-team/ionic-framework/issues/24181#issuecomment-971550734

This is not really a specific Test Utils problem, but more of how jsdom chooses to implement their attribute serializer, but it only manifests when using shallowMount. There's not a real "fix", but we will probably need some kind of hack/escape hatch for this scenario (stubs w/ symbols). See my link ^ for more.

**Comment by boboldehampsink** [comment]
@lmiller1990 any news? Ionic 6.0 is now out as stable, but can't be used with Vue before this is fixed.

**Comment by lmiller1990** [comment]
@boboldehampsink Ionic 6 seems to use Vue 3, so you'll need Test Utils v2. This bug is fixed here https://github.com/vuejs/vue-test-utils-next/issues/1076 which will be released this week.

As for back-porting that fix, you could do something similar and strip out symbols in this library. It's not a priority for me right now, but if you'd like to make a PR, that'd be great.

**Comment by ebisbe** [comment]
@lmiller1990 Do we leave this ticket open for back porting it? 
For me, no response from 2y so I'd go and close it. If anyone is interested he can make a PR and back port it, right?

**Comment by lmiller1990** [comment]
I think it's fine to leave this open - known issue, not fixed. It's unlikely anyone is using Vue 2 with Ionic now, they would likely be looking to upgrade.

**Comment by wangxiang4** [comment]
This problem can be solved using the wrapping layer arrow function

**Comment by lmiller1990** [comment]
Please share a snippet @wangxiang4, this sounds like useful info. Thanks!
